### PR TITLE
docs: check and improve handling of duplicate tags

### PR DESCRIPTION
**Pull request by ST-DDT**
~~Ref: #2439~~

- ~~#2439~~

---

Preparation work for enableing the [`unicorn/require-array-join-separator`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/require-array-join-separator.md) lint rule.

### CODE REVIEW

**Comment by matthewmayer** [comment]
This is a behavior change no? The default for join is ,

**Comment by ST-DDT** [comment]
Technically yes. Practically not, because we only ever have one.

**Comment by matthewmayer** [comment]
If we did have more than one should that throw an error?

**Comment by ST-DDT** [comment]
> If we did have more than one should that throw an error?

Changed. It looks like there was one method with two `@throws` tags, that I have missed in my earlier checks.

**Comment by matthewmayer** [comment]
This PR should probably be renamed now then

**Comment by ST-DDT** [comment]
> This PR should probably be renamed now then

The title is already very long so appending it might be hard.
Do you have a suggestion?

**Comment by matthewmayer** [comment]
Split the changes about allowing one one of each tag to a seperate PR?

**Comment by ST-DDT** [comment]
IMO this only does a single thing, what should I split here?

I renamed it to reflect its doc impact

**Comment by xDivisionByZerox** [comment]
Preview of the changes on multiple `@throw` tags:

image

---

I don't know if I'm the biggest fan as you can't tell right now that these are 2 distinct throw situations. But it is better than not having them at all.

**Comment by fisker** [comment]
This end up with nothing related to `require-array-join-separator` at all... LOL

 `require-array-join-separator` mean to make code easier for beginners and for consistency.

**Comment by xDivisionByZerox** [comment]
> This end up with nothing related to `require-array-join-separator` at all... LOL
> 
>  `require-array-join-separator` mean to make code easier for beginners and for consistency.

True. Enabling the rule in this PR is kind of missleading.

File: .eslintrc.js [hunk 1]
```diff
@@ -62,7 +62,6 @@ module.exports = defineConfig({
     'unicorn/prefer-ternary': 'off',
     'unicorn/prefer-top-level-await': 'off',
     'unicorn/prevent-abbreviations': 'off',
```
Comment (ST-DDT) [inline .eslintrc.js]:
```suggestion
    'unicorn/prevent-abbreviations': 'off',
    'unicorn/require-array-join-separator': 'off',
```

File: scripts/apidoc/typedoc.ts [hunk 2]
```diff
@@ -382,5 +414,5 @@ export function extractThrows(reflection?: CommentHolder): string | undefined {
  * @returns The contents of the `@since` tag.
  */
 export function extractSince(reflection: CommentHolder): string {
```
Comment (xDivisionByZerox) [inline scripts/apidoc/typedoc.ts:416]: How can this return `string` if `extractSingleTagContent` returns `string | undefined`? 🤔

Comment (ST-DDT) [reply scripts/apidoc/typedoc.ts:416]: Because https://github.com/faker-js/faker/pull/2435 is not yet merged.

- #2435

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**